### PR TITLE
[ARKit] Add support for Xcode 12 beta 1.

### PR DESCRIPTION
**Pull request by mandel-macaque**
PInvoke that is missing is due to a dependency with Visio that is not
yet updated.

### CODE REVIEW

File: src/arkit.cs [hunk 1]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
 		[NullAllowed]
 		ARGeometrySource Classification { get; }
 	}
+
+	[iOS (14,0)]
+	[BaseType (typeof(NSObject))]
```
Comment (mandel-macaque) [inline src/arkit.cs:2207]:
```suggestion
	[BaseType (typeof (NSObject))]
```

File: src/arkit.cs [hunk 2]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
...
+	interface ARDepthData {
+		[Export ("depthMap", ArgumentSemantic.Assign)]
+		CVPixelBuffer DepthMap { get; }
+
+		[NullAllowed, Export ("confidenceMap", ArgumentSemantic.Assign)]
+		CVPixelBuffer ConfidenceMap { get; }
+	}
+
+	[iOS (14,0)]
+	[BaseType (typeof(ARAnchor))]
```
Comment (mandel-macaque) [inline src/arkit.cs:2217]:
```suggestion
	[BaseType (typeof (ARAnchor))]
```

File: src/arkit.cs [hunk 3]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
...
+
+		[Export ("initWithName:coordinate:")]
+		IntPtr Constructor (string name, CLLocationCoordinate2D coordinate);
+
+		[Export ("initWithName:coordinate:altitude:")]
+		IntPtr Constructor (string name, CLLocationCoordinate2D coordinate, double altitude);
+	}
+
+	[iOS (14,0)]
+	[BaseType (typeof(ARConfiguration))]
```
Comment (mandel-macaque) [inline src/arkit.cs:2247]:
```suggestion
	[BaseType (typeof (ARConfiguration))]
```

File: src/arkit.cs [hunk 4]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
...
+		void CheckAvailability (CLLocationCoordinate2D coordinate, Action<bool, NSError> completionHandler);
+
+		[Static]
+		[Export ("new")]
+		[return: Release]
+		ARGeoTrackingConfiguration Create ();
+	}
+
+	[iOS (14,0)]
+	[BaseType (typeof(NSObject))]
```
Comment (mandel-macaque) [inline src/arkit.cs:2291]:
```suggestion
	[BaseType (typeof (NSObject))]
```

File: src/arkit.cs [hunk 5]
```diff
@@ -800,6 +862,12 @@ interface ARSession {
 		[iOS (13,0)]
 		[Export ("updateWithCollaborationData:")]
 		void Update (ARCollaborationData collaborationData);
+
+		[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:866]:
```suggestion
		[iOS (14, 0)]
```

File: src/arkit.cs [hunk 6]
```diff
@@ -829,6 +897,10 @@ interface ARSessionObserver {
 		[iOS (13,0)]
 		[Export ("session:didOutputCollaborationData:")]
 		void DidOutputCollaborationData (ARSession session, ARCollaborationData data);
+
+		[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:901]:
```suggestion
		[iOS (14, 0)]
```

File: src/arkit.cs [hunk 7]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
 		[NullAllowed]
 		ARGeometrySource Classification { get; }
 	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:2206]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 8]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
...
+	[BaseType (typeof(NSObject))]
+	interface ARDepthData {
+		[Export ("depthMap", ArgumentSemantic.Assign)]
+		CVPixelBuffer DepthMap { get; }
+
+		[NullAllowed, Export ("confidenceMap", ArgumentSemantic.Assign)]
+		CVPixelBuffer ConfidenceMap { get; }
+	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:2216]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 9]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
...
+		IntPtr Constructor (CLLocationCoordinate2D coordinate, double altitude);
+
+		[Export ("initWithName:coordinate:")]
+		IntPtr Constructor (string name, CLLocationCoordinate2D coordinate);
+
+		[Export ("initWithName:coordinate:altitude:")]
+		IntPtr Constructor (string name, CLLocationCoordinate2D coordinate, double altitude);
+	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:2246]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 10]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
...
+		[Export ("checkAvailabilityAtCoordinate:completionHandler:")]
+		void CheckAvailability (CLLocationCoordinate2D coordinate, Action<bool, NSError> completionHandler);
+
+		[Static]
+		[Export ("new")]
+		[return: Release]
+		ARGeoTrackingConfiguration Create ();
+	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:2290]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 11]
```diff
@@ -232,6 +233,55 @@ public enum ARCollaborationDataPriority : long {
...
+	[Native]
+	public enum ARGeoTrackingState : long {
+		NotAvailable,
+		Initializing,
+		Localizing,
+		Localized,
+	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:272]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 12]
```diff
@@ -232,6 +233,55 @@ public enum ARCollaborationDataPriority : long {
...
+	[Native]
+	public enum ARGeoTrackingAccuracy : long {
+		Undetermined,
+		Low,
+		Medium,
+		High,
+	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:263]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 13]
```diff
@@ -232,6 +233,55 @@ public enum ARCollaborationDataPriority : long {
...
+	[Native]
+	public enum ARConfidenceLevel : long {
+		Low,
+		Medium,
+		High,
+	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:254]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 14]
```diff
@@ -232,6 +233,55 @@ public enum ARCollaborationDataPriority : long {
...
+	[Native]
+	public enum ARAltitudeSource : long {
+		Unknown,
+		Coarse,
+		Precise,
+		UserDefined,
+	}
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:246]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 15]
```diff
@@ -232,6 +233,55 @@ public enum ARCollaborationDataPriority : long {
 		Optional,
 	}
 
+
+	[iOS (14,0)]
```
Comment (mandel-macaque) [inline src/arkit.cs:237]:
```suggestion
	[iOS (14, 0)]
```

File: src/arkit.cs [hunk 16]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
...
+	interface ARGeoTrackingConfiguration {
+		[Static]
+		[Export ("supportedVideoFormats")]
+		ARVideoFormat[] GetSupportedVideoFormats ();
+
+		[Export ("environmentTexturing", ArgumentSemantic.Assign)]
+		AREnvironmentTexturing EnvironmentTexturing { get; set; }
+
+		[Export ("wantsHDREnvironmentTextures")]
+		bool WantsHDREnvironmentTextures { get; set; }
```
Comment (rolfbjarne) [inline src/arkit.cs:2257]: We already have this as `Hdr`: https://github.com/xamarin/xamarin-macios/blob/46b6d3787d50c4bf8bbbade171d6025f8edea602/src/arkit.cs#L1724-L1725

```suggestion
		bool WantsHdrEnvironmentTextures { get; set; }
```

File: src/arkit.cs [hunk 17]
```diff
@@ -800,6 +862,12 @@ interface ARSession {
 		[iOS (13,0)]
 		[Export ("updateWithCollaborationData:")]
 		void Update (ARCollaborationData collaborationData);
+
+		[iOS (14, 0)]
+		[Async (ResultTypeName="GeoLocationForPoint")]
+		[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+		[Export ("getGeoLocationForPoint:completionHandler:")]
+		void GetGeoLocation (Vector3 position, Action<CLLocationCoordinate2D, double, NSError> completionHandler);
```
Comment (spouliot) [inline src/arkit.cs:870]: action with more than 1 (and NSError) arguments should be a custom delegate so the argument names serves as self-documentation

File: src/arkit.cs [hunk 18]
```diff
@@ -2086,4 +2202,103 @@ interface ARMeshGeometry : NSSecureCoding {
 		[NullAllowed]
 		ARGeometrySource Classification { get; }
 	}
+
+	[iOS (14, 0)]
+	[BaseType (typeof (NSObject))]
```
Comment (spouliot) [inline src/arkit.cs:2207]: `[DisableDefaultCtor]` as this is only exposed in a getter so there's no reason to be able to create one (if it even works)

**Pull request merged**